File: 3016. Minimum Number of Pushes to Type Word II/Solution.py
class Solution(object):
    def minimumPushes(self, word):
        """
        :type word: str
        :rtype: int
        """

        cfreq = {}

        for ch in word:
            cfreq[ch] = cfreq.get(ch,0)+1

        freq = sorted(cfreq.values(),reverse = True)

        res = 0

        for i,f in enumerate(freq):
            if i<8:
                res += f*1

            elif i<16:
                res += f*2

            elif i<24:
                res += f*3

            else :
                res += f*4

        return res


        

File: 3016. Minimum Number of Pushes to Type Word II/Solution.java
class Solution {
    public int minimumPushes(String word) {
        HashMap<Character,Integer> cfreq = new HashMap<>();

        for(char ch : word.toCharArray()){
            cfreq.put(ch,cfreq.getOrDefault(ch,0)+1);
        }

        List<Integer> freq =  new ArrayList<>(cfreq.values());

        Collections.sort(freq,Collections.reverseOrder());

        int res = 0;

        for(int i=0;i<freq.size();i++){
            int f = freq.get(i);

            if(i<8){
                res+=f;
            }
            else if(i<16){
                res+= 2*f;
            }
            else if(i<24){
                res+= 3*f;
            }
            else{
                res+= 4*f;
            }
        }

        return res;
    }
}